### ISSUE TITLE
Bug: Unwanted Key Stored in Redis

### ISSUE THREAD

**Issue by x-sushant-x** [x-sushant-x/Rate-Shield#1]
### Description
When processing a request, two keys are being stored in Redis: one with the format ip:endpoint (e.g., 127.0.0.1:/api/v1/get-data) and another with just the IP address (e.g., 127.0.0.1). The expected behavior is that only the ip:endpoint key should be stored.

### Steps to Reproduce
1. Send a request to the rate limiter API with the following headers:
     ```ip: 127.0.0.1```
     ```endpoint: /api/v1/get-data```
2. Check the keys stored in Redis.
3. Expected Behaviour
Only one key should be stored in Redis: ```127.0.0.1:/api/v1/get-data```

#### Actual Behavior
Two keys are stored in Redis:

1. 127.0.0.1:/api/v1/get-data
2. 127.0.0.1

**Comment by figuerom16** [comment]
It looks like the key isn't being completed with the endpoint before saving.
```go
func (t *TokenBucketService) createBucket(ip, endpoint string, capacity, tokenAddRate int) models.Bucket {
	bucket := models.Bucket{
		ClientIP:        ip,
		CreatedAt:       time.Now().Unix(),
		Capacity:        capacity,
		AvailableTokens: capacity,
		Endpoint:        endpoint,
		TokenAddRate:    tokenAddRate,
		TokenAddTime:    DefaultTokenAddTime,
	}

	t.saveBucket(ip, bucket)
	return bucket
}
```

Should this be...
```go
func (t *TokenBucketService) createBucket(ip, endpoint string, capacity, tokenAddRate int) models.Bucket {
	...
	t.saveBucket(ip + ":" endpoint, bucket)
	return bucket
}
```

**Comment by figuerom16** [comment]
Thinking about this more.  saveBucket should be paramless to avoid these problems.  I'd make createBucket a basic function then instead of t.saveBucket you can call bucket.saveBucket() since all the necessary information is part of bucket model.

```go
func (t *TokenBucketService) saveBucket() error {
	key := t.ClientIP + ":" + t.Endpoint
	if err := redisClient.SetJSONObject(key, t); err != nil {
		log.Error().Err(err).Msg("Error saving new bucket to Redis")
		return err
	}

	if err := redisClient.TokenBucketClient.Expire(context.Background(), key, BucketExpireTime).Err(); err != nil {
		log.Error().Err(err).Msg("Error setting bucket expiration in Redis")
		return err
	}

	return nil
}
```

**Comment by x-sushant-x** [comment]
Cool! I agree with you. Do you wanna take this issue and open a PR?

**Comment by x-sushant-x** [comment]
Issue solved via #3
